data-prepper-plugins/http-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/configuration/HttpSinkConfigurationTest.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.sink.configuration;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import org.hamcrest.CoreMatchers;
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.model.types.ByteCount;
+import software.amazon.awssdk.regions.Region;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class HttpSinkConfigurationTest {
+
+    private static final String SINK_YAML = "        urls:\n" +
+            "        - url: \"https://httpbin.org/post\"\n" +
+            "          workers: 1\n" +
+            "          proxy: test\n" +
+            "          codec:\n" +
+            "            ndjson:\n" +
+            "          http_method: \"POST\"\n" +
+            "          auth_type: \"http_basic\"\n" +
```

Comment:
What is the reason behind having both this `auth_type` and the `authentication` with `http_basic`? Do we need both?

Reply:
Since there will be two kinds of authentication for this plugin, basic and bearer token, so **auth_type** for every url will signify and kind of auth and **authentication** section will have the details of it eg:

authentication:
          http_basic:
            username: "admin"
            password: "admin"
          bearer_token:
            token: ""